### ISSUE TITLE
Add a mkosi.init script to run before configuration files parsing happens

### ISSUE THREAD

**Issue by davide125** [systemd/mkosi#3176]
### mkosi commit the issue has been seen with

main

### Used host distribution

Fedora Linux 41

### Used target distribution

Fedora Linux 41

### CPU architectures issue was seen on

None

### Unexpected behaviour you saw

I'd like to have a `mkosi.init` script to run at the very beginning (before config parsing/validation) in order to create/initialize the contents of things like `PackageDirectories`  and `ExtraSearchPaths`. The specific usecase I have in mind is building systemd as part of the mkosi build, and including the systemd packages into the image built by mkosi. My attempt at that involved cloning systemd in the mkosi directory, and then adding a `mkosi.sync` with `mkosi --directory=systemd --release=41 -t none` to build it. However, this runs too late, and mkosi ends up failing because the build directory doesn't exist yet.

### Used mkosi config

```ini
[Distribution]
Distribution=fedora
Release=41

[Content]
PackageDirectories=systemd/build/mkosi.builddir/%d~%r~%a

[Host]
ExtraSearchPaths=systemd/build/mkosi.builddir/%d~%r~%a
```

### mkosi output

```sh
‣ systemd/build/mkosi.builddir/fedora~41~x86-64 does not exist
```

**Comment by davide125** [comment]
Obviously, a workaround for this is manually running `mkdir systemd/build/mkosi.builddir/fedora~41~x86-64` before attempting to build, but that's an additional manual step I'd like to avoid if possible.

**Comment by DaanDeMeyer** [comment]
So I'm conflicted on this one but I also think this is a bit too niche to warrant introducing an extra script.

I'd suggest reworking your setup to not build systemd in mkosi.sync but instead have a wrapper script that builds systemd and then execs into mkosi, and passthrough all arguments to that script to mkosi. Then you keep full flexibility of passing CLI options to mkosi, and the only con is that you have to invoke a separate script instead of mkosi directly, but this is the same limitation as in the systemd repository, where you have to run `mkosi sandbox meson compile -C build` to build systemd on the host first before running mkosi.
